src/main/java/com/cburch/logisim/std/memory/DualportRam.java

```diff
@@ -0,0 +1,423 @@
+/*
+ * Logisim-evolution - digital logic design tool and simulator
+ * Copyright by the Logisim-evolution developers
+ *
+ * https://github.com/logisim-evolution/
+ *
+ * This is free software released under GNU GPLv3 license
+ */
+
+package com.cburch.logisim.std.memory;
+
+import com.cburch.logisim.circuit.Circuit;
+import com.cburch.logisim.circuit.CircuitState;
+import com.cburch.logisim.comp.Component;
+import com.cburch.logisim.data.*;
+import com.cburch.logisim.fpga.designrulecheck.CorrectLabel;
+import com.cburch.logisim.fpga.designrulecheck.netlistComponent;
+import com.cburch.logisim.gui.hex.HexFrame;
+import com.cburch.logisim.gui.icons.ArithmeticIcon;
+import com.cburch.logisim.instance.*;
+import com.cburch.logisim.proj.Project;
+
+import java.util.WeakHashMap;
+import java.util.function.Consumer;
+
+import static com.cburch.logisim.std.Strings.S;
+
+public class DualportRam extends Mem {
```

Comment:
I am not a 100 percent sure about the camel-case conventions, but `DualPortRam` might be more consistent with it. If you should agree, please apply the name change analogously to the other classes.